packages/cli/src/utils/api.js

```diff
@@ -449,17 +452,42 @@ const upload = async (app, { skipValidation = false } = {}) => {
   const binarySourceZip = fs.readFileSync(fullSourceZipPath);
   const sourceBuffer = Buffer.from(binarySourceZip).toString('base64');
 
+  const headers = {};
+  if (overwritePartnerChanges) {
+    headers['X-Overwrite-Partner-Changes'] = 'true';
+  }
+
   startSpinner(`Uploading version ${definition.version}`);
-  await callAPI(`/apps/${app.id}/versions/${definition.version}`, {
-    method: 'PUT',
-    body: {
-      zip_file: buffer,
-      source_zip_file: sourceBuffer,
-      skip_validation: skipValidation,
-    },
-  });
+  try {
+    await callAPI(
+      `/apps/${app.id}/versions/${definition.version}`,
+      {
+        method: 'PUT',
+        body: {
+          zip_file: buffer,
+          source_zip_file: sourceBuffer,
+          skip_validation: skipValidation,
+        },
+        extraHeaders: headers,
+      },
+      true,
```

Comment:
This `true` adds `rawError = True` so we can catch a response as `err` and can do `err.status`.